### PR TITLE
Improve UX of Component Browser

### LINKED ISSUE #6486: CB is adding `Column.from_vector` when creating a vector
### Discord username

_No response_

### What type of issue is this?

Permanent – Occurring repeatably

### Is this issue blocking you from using Enso?

- [ ] Yes, I can't use Enso because of this issue.

### Is this a regression?

- [ ] Yes, previous version of Enso did not have this issue.

### What issue are you facing?

When typing a vector the CB is always adding `Column.from_vector` after closing `]` when pressing `enter`

### Expected behaviour

Vector should be treated like a number or literal - as a special input type

### How we can reproduce it?

_No response_

### Screenshots or screencasts

_No response_

### Enso Version

develop @ 6b0c682b08b7745368cec3474cc774c51a5b48f5

### Browser or standalone distribution

Standalone distribution

### Browser Version or standalone distribution

standalone

### Operating System

MacOS

### Operating System Version

_No response_

### Hardware you are using

_No response_

### LINKED ISSUE #6601: Problems when comitting string literal edit with the enter key.
Do the following:
1. Start a new project.
2. Ctrl+Click on the initial node to edit the literal.
3. Place the edit caret after the space character. 

Now, either:
4. Press the Backspace key to erase the space. 
5. Press the Return key to commit the change.

The node reverts to the original literal text (space comes back).

Alternatively:
4. Press the Space key to insert one more (second) space character.
5. Press the Return key to commit the change.

Now the text literal ends up with 3 space characters in that place. 

https://user-images.githubusercontent.com/1548407/236852729-bc5081e8-aecc-452c-b105-f3c12e0478b5.mp4

### LINKED ISSUE #6736: Confusing preview output of vector method
### Discord username

_No response_

### What type of issue is this?

Permanent – Occurring repeatably

### Is this issue blocking you from using Enso?

- [ ] Yes, I can't use Enso because of this issue.

### Is this a regression?

- [ ] Yes, previous version of Enso did not have this issue.

### What issue are you facing?

While learning Enso I get confused by the preview of method I'm exploring being what I'd expect as a result of the method to later find out that it's not working.

https://github.com/enso-org/enso/assets/108271897/60506f6b-2b71-4a85-b722-c36cac416d02

I wanted to create vector from a table / colum and got a decent result in a preview...

### Expected behaviour

If the method isn't correct I'd like to be informed about it with possible solution suggested.

### How we can reproduce it?

_No response_

### Screenshots or screencasts

_No response_

### Enso Version

2023.5.17 nightly

### Browser or standalone distribution

Standalone distribution

### Browser Version or standalone distribution

standalone

### Operating System

Windows

### Operating System Version

Win11pro 22H2 22621.1555

### Hardware you are using

12th Gen Intel(R) Core(TM) i9-12900HK / RTX3060 Laptop / Nvidia Drivers 531.68

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description

Fixes #6736 
Fixes #6486
Fixes #6601 

To achieve goal mentioned in the title the following changes were made:
1. Now it's normal state of Component Browser to have no entry selected. In that case pressing enter will accept the input as-is.
2. The issues of `Column.from_vector` or actually any code being unintentionally inserted should be resolved, as in most of those scenarios we do not select any item in CB after list reload. Specifically, the first element is selected only when:
   * the input is empty (so the user just starts writing code)
   * or we filter by some pattern (we select the best match). This includes writing literal.
3. On this occasion, inserting literal was fixed, so no more random spaces are inserted inside.
4. Also, the entries in groups other than local group are reversed: now the entries specified as first in "component groups" section in the library will be easily approachable. In particular, now "Text input" is selected by default in empty node instead of `Column.from_vector`.

https://github.com/enso-org/enso/assets/3919101/42064b08-16b4-4519-81de-8a2d3701c112

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/gui/src/controller/graph/executed.rs [hunk 1]
```diff
@@ -233,7 +233,7 @@ impl Handle {
             let mut successful_calls = Vec::new();
             let result = stream::iter(stack)
                 .then(|local_call| async {
-                    debug!("Entering node {}.", local_call.call);
+                    warn!("Entering node {}.", local_call.call);
```
Comment (MichaelMauderer) [inline app/gui/src/controller/graph/executed.rs]: Debug logging.

File: app/gui/src/controller/searcher.rs [hunk 2]
```diff
@@ -755,43 +782,54 @@ impl Searcher {
     /// expression, otherwise a new node is added. This will also add all imports required by
     /// picked suggestions.
     #[profile(Debug)]
-    pub fn commit_node(&self) -> FallibleResult<ast::Id> {
+    pub fn commit_node(&self) -> FallibleResult<Option<ast::Id>> {
```
Comment (MichaelMauderer) [inline app/gui/src/controller/searcher.rs]: Some docs about the return type might be useful here. What does it mean to succeed with no node (`Ok(None)`)? While this is explained in the code below, this should also be in the method doc.

File: app/gui/src/controller/searcher.rs [hunk 3]
```diff
@@ -1845,14 +1883,14 @@ pub mod test {
...
             // The input was manually written (not picked).
-            Case::new(
-                "2 + 2",
-                &["sum1 = 2 + 2", "operator1 = sum1.testFunction1"],
-                false,
-                |f, _| {
-                    f.searcher.set_input("testFunction1".to_owned(), Byte(13)).unwrap();
-                },
-            ),
+            // Case::new(
```
Comment (MichaelMauderer) [inline app/gui/src/controller/searcher.rs]: Commented test.

File: app/gui/src/controller/searcher/input.rs [hunk 4]
```diff
@@ -636,7 +652,7 @@ mod tests {
             }
 
             fn run(self, parser: &Parser) {
-                debug!("Running case {} cursor position {}", self.input, self.cursor_position);
+                println!("Running case {} cursor position {}", self.input, self.cursor_position);
```
Comment (MichaelMauderer) [inline app/gui/src/controller/searcher/input.rs]: Changed logging.

**Review by MichaelMauderer — APPROVED** [review]

File: app/gui/src/controller/searcher.rs (hunk 2 above)
Comment (farmaazon) [reply app/gui/src/controller/searcher.rs]: Now I think it should not be handled here. Commit node should not treat "non-updating" node as a normal routine. Instead, the presenter is handling that case.

**Review by wdanilo — APPROVED** [review]
Beautiful!

**Comment by MichaelMauderer** [comment]
QA :red_circle:

I just did a quick test and it appears that I can no longer create a new node with a number literal. The same happens for any input and also for selections from the CB, via mouse and keyboard.

In the video, I did the following things:
* Press tab
* Press a random string of numbers
* Press enter
After that, the CB closes and the new node disappears. 

https://github.com/enso-org/enso/assets/1428930/997ac3c5-2267-4744-b58c-0831f4a89abd

**Comment by farmaazon** [comment]
@MichaelMauderer ready for re-review.

**Comment by MichaelMauderer** [comment]
:yellow_circle:  Looks mostly good to me. There is one issue, however, that I noticed:

When pressing tab to create a new node, and pressing tab again, a `""` is inserted. Previously, this would insert the selected CB entry, but I assume since nothing is selected, “nothing” gets inserted, but this should probably not be an empty string literal.

**Comment by farmaazon** [comment]
> Previously, this would insert the selected CB entry, but I assume since nothing is selected, “nothing” gets inserted, but this should probably not be an empty string literal.

Actually, it _does_ select the selected entry (as before) - just now (due to reverse order of elements - see point 4.) the first element is "Text literal", not "Column.from_vector"

**Comment by MichaelMauderer** [comment]
Ah you are right. So that is all fine then: :green_apple: :green_book: :green_heart:

**Pull request merged**